torchdynamo: add convolution pointwise(unary) fusion kernel

XiaobingSuper opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #86583
* #86582
* __->__ #86581

Support unary fusion of Convolution with:

- relu
- sigmoid
- tanh
- hardswish
- leaky_relu
- hardtanh
- gelu

XiaobingSuper committed:
torchdynamo: add convolution eltwise fusion kernel

[ghstack-poisoned]

XiaobingSuper committed:
Update on "torchdynamo: add convolution eltwise fusion kernel"




[ghstack-poisoned]

jgong5 commented on test/test_mkldnn_fusion.py:
```diff
@@ -13,7 +13,6 @@
 FUSION_GROUP = 'prim::TensorExprGroup'
 
-
 class EltwiseFusionOp(NamedTuple):
```
`PointwisePostOp` is more accurate?

XiaobingSuper replied:
Yes, I change it.

XiaobingSuper committed:
Update on "torchdynamo: add convolution pointwise(unary) fusion kernel"


Support unary fusion of Convolution with:

- relu
- sigmoid
- tanh
- hardswish
- leaky_relu
- hardtanh
- gelu


[ghstack-poisoned]

XiaobingSuper commented on aten/src/ATen/native/mkldnn/Conv.cpp:
```diff
@@ -244,6 +258,43 @@ Tensor mkldnn_convolution(
+    IntArrayRef stride,
+    IntArrayRef dilation,
+    int64_t groups,
+    std::string attr,
```
Yes, changed.

XiaobingSuper commented on aten/src/ATen/native/mkldnn/Conv.cpp:
```diff
@@ -244,6 +258,43 @@ Tensor mkldnn_convolution(
+    int64_t groups,
+    std::string attr,
+    std::vector<c10::optional<at::Scalar>> scalars,
+    c10::optional<std::string> algorithm) {
```
changed.

XiaobingSuper commented on aten/src/ATen/native/mkldnn/Conv.cpp:
```diff
@@ -244,6 +258,43 @@ Tensor mkldnn_convolution(
+    IntArrayRef dilation,
+    int64_t groups,
+    std::string attr,
+    std::vector<c10::optional<at::Scalar>> scalars,
```
done.

XiaobingSuper committed:
Update on "torchdynamo: add convolution pointwise(unary) fusion kernel"


Support unary fusion of Convolution with:

- relu
- sigmoid
- tanh
- hardswish
- leaky_relu
- hardtanh
- gelu


[ghstack-poisoned]

EikanWang commented:
@jansel , could you help review this PR for TorchInductor?

EikanWang commented:
@jansel , I noticed that you approved the other two PRs. May I know if you have any concerns about this PR?

jansel commented:
> @jansel , I noticed that you approved the other two PRs. May I know if you have any concerns about this PR?

Nope just missed it.  Sorry, too many notifications!

EikanWang commented:
@pytorchbot merge

EikanWang commented:
@XiaobingSuper , could you please rebase the PR and resolve the conflict?

XiaobingSuper committed:
Update on "torchdynamo: add convolution pointwise(unary) fusion kernel"


Support unary fusion of Convolution with:

- relu
- sigmoid
- tanh
- hardswish
- leaky_relu
- hardtanh
- gelu


[ghstack-poisoned]

XiaobingSuper commented:
@EikanWang , code rebased.

XiaobingSuper committed:
Update on "torchdynamo: add convolution pointwise(unary) fusion kernel"


Support unary fusion of Convolution with:

- relu
- sigmoid
- tanh
- hardswish
- leaky_relu
- hardtanh
- gelu


[ghstack-poisoned]

EikanWang commented:
@pytorchbot merge
